crates/ide-completion/src/completions/dot.rs

```diff
@@ -35,11 +40,33 @@ pub(crate) fn complete_dot(
         );
         item.detail("expr.await");
         item.add_to(acc, ctx.db);
-    }
 
-    let is_field_access = matches!(dot_access.kind, DotAccessKind::Field { .. });
-    let is_method_access_with_parens =
-        matches!(dot_access.kind, DotAccessKind::Method { has_parens: true });
+        // Completions that skip `.await`, e.g. `.await.foo()`.
+        let dot_access_kind = match &dot_access.kind {
+            DotAccessKind::Field { receiver_is_ambiguous_float_literal: _ } => {
+                DotAccessKind::Field { receiver_is_ambiguous_float_literal: false }
+            }
+            it @ DotAccessKind::Method { .. } => *it,
+        };
```

Comment:
```suggestion
        let dot_access_kind = match &dot_access.kind {
            DotAccessKind::Field { receiver_is_ambiguous_float_literal: _ } => {
                DotAccessKind::Field { receiver_is_ambiguous_float_literal: false }
            }
            it => *it,
        };
```
Alternatively clone, then mutably if let the first variant to overwrite the field?

Reply:
I want it to fail compilation if we add a variant that may have data that needs to be updated, although either is fine.